controllers/argocd/applicationset.go

```diff
@@ -144,9 +149,26 @@ func (r *ReconcileArgoCD) reconcileApplicationSetDeployment(cr *argoproj.ArgoCD,
 			},
 		},
 	}
+	addSCMGitlabVolumeMount := false
+	if cr.Spec.ApplicationSet.SCMRootCAPath != "" {
```

Comment:
Why should we get the CA path from the user, if the operator is going to mount it from a well defined CM. Can the operator not create an empty config map `argocd-appset-gitlab-scm-tls-certs-cm` and mount it in a well defined path like `/app/tls/cert` and if at all user wants to use this feature, all he has to do is to fill Certificate date in config map `argocd-appset-gitlab-scm-tls-certs-cm`. It would be easy for the operator to manage the `Volume` and `VolumeMounts` as its no longer conditonal elements. Since the mount path is well defined, there won't be any path restrictions that user has to be aware of.

Reply:
The alternate approach, would be similar to how TLS certs are handled in ArgoCD Server.

```
oc get deploy -n openshift-gitops openshift-gitops-server -o yaml
....
- mountPath: /app/config/tls
          name: tls-certs
....
volumes:
      - configMap:
          defaultMode: 420
          name: argocd-tls-certs-cm
        name: tls-certs
```

```
oc get cm -n openshift-gitops argocd-tls-certs-cm -o yaml

apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-tls-certs-cm
  namespace: openshift-gitops
```

Reply:
We would not want to create an empty CM as the requests for Gitlab client would use these certs only if the parameter `--scm-root-ca-path` is set on the ApplicationController pod. And, we would want to set this path only if user has provided the ConfigMap.

I do like the thought of hard-coding the path of the cert, but I am wondering if there would be any use case where user would want to specify the path. I am open to hard-coding the path for volume mount.

Reply:
Yes, we can do that. I will update the code.

Reply:
The one you had suggested in the example, `/app/tls/cert` looks good to me. If we want to give some kind of user configurable option, to explicitly enable this feature, then we can consider taking in the config map name in the ArgoCD spec and keep the mount path hard coded to `/app/tls/cert`

Reply:
/lgtm